programs/intent-transfer/src/lib.rs

```diff
@@ -576,6 +630,32 @@ fn verify_and_update_nonce(nonce: &mut Account<Nonce>, new_nonce: u64) -> Result
     Ok(())
 }
 
+fn verify_ntt_manager(ntt_manager_key: Pubkey, expected_ntt_config: &Account<'_, ExpectedNttConfig>) -> Result<()> {
+    require_keys_eq!(
+        ntt_manager_key,
+        expected_ntt_config.manager,
+        IntentTransferError::InvalidNttManager
+    );
+    Ok(())
+}
+
+fn verify_upgrade_authority(
```

Comment:
nit: could write these as 
   `#[account(address = bpf_loader_upgradeable::get_program_data_address(&crate::ID))]`
and:
    `#[account(mut, address = program_data.upgrade_authority_address.ok_or(IntentTransferError::Unauthorized)?)]`